hpx/lcos/detail/future_data.hpp

```diff
@@ -350,14 +350,6 @@ namespace detail
         // allowed
         void handle_on_completed(completed_callback_type && on_completed)
         {
-            // We need to run the completion asynchronously if we aren't on a
-            // HPX thread
-            if (HPX_UNLIKELY(nullptr == threads::get_self_ptr()))
-            {
-                HPX_THROW_EXCEPTION(null_thread_id,
-                        "future_data::handle_on_completed",
-                        "null thread id encountered");
-            }
```

Comment:
This check is really only needed if we need to schedule the completion handler on a new thread.